# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../unionfs/linux-2.6.19.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---


Author: Shaya Potter spotter at cs.columbia.edu 
[Unionfs] get unionfs building on 2.6.19.1
http://www.fsl.cs.sunysb.edu/pipermail/unionfs/2007-January/004989.html

diff -rNu unionfs-1.5pre-cvs200612261700/dirfops.c unionfs-1.5pre-cvs200612261700.good/dirfops.c
--- unionfs-1.5pre-cvs200612261700/dirfops.c	2006-12-26 22:00:45.000000000 +0000
+++ unionfs-1.5pre-cvs200612261700.good/dirfops.c	2007-01-02 22:57:44.000000000 +0000
@@ -41,7 +41,7 @@
 
 /* copied from generic filldir in fs/readir.c */
 static int unionfs_filldir(void *dirent, const char *name, int namelen,
-			   loff_t offset, ino_t ino, unsigned int d_type)
+			   loff_t offset, u64 ino, unsigned int d_type)
 {
 	struct unionfs_getdents_callback *buf =
 	    (struct unionfs_getdents_callback *)dirent;
diff -rNu unionfs-1.5pre-cvs200612261700/dirhelper.c unionfs-1.5pre-cvs200612261700.good/dirhelper.c
--- unionfs-1.5pre-cvs200612261700/dirhelper.c	2006-12-26 22:00:45.000000000 +0000
+++ unionfs-1.5pre-cvs200612261700.good/dirhelper.c	2007-01-02 22:57:07.000000000 +0000
@@ -155,7 +155,7 @@
 
 /* This filldir function makes sure only whiteouts exist within a directory. */
 static int readdir_util_callback(void *dirent, const char *name, int namelen,
-				 loff_t offset, ino_t ino, unsigned int d_type)
+				 loff_t offset, u64 ino, unsigned int d_type)
 {
 	int err = 0;
 	struct unionfs_rdutil_callback *buf =
diff -rNu unionfs-1.5pre-cvs200612261700/lookup.c unionfs-1.5pre-cvs200612261700.good/lookup.c
--- unionfs-1.5pre-cvs200612261700/lookup.c	2006-12-26 22:00:45.000000000 +0000
+++ unionfs-1.5pre-cvs200612261700.good/lookup.c	2007-01-02 22:57:21.000000000 +0000
@@ -399,9 +399,8 @@
 {
 	if (!unionfs_dentry_cachep)
 		return;
-	if (kmem_cache_destroy(unionfs_dentry_cachep))
-		printk(KERN_ERR
-		       "unionfs_dentry_cache: not all structures were freed\n");
+	kmem_cache_destroy(unionfs_dentry_cachep);
+
 	return;
 }
 
diff -rNu unionfs-1.5pre-cvs200612261700/rdstate.c unionfs-1.5pre-cvs200612261700.good/rdstate.c
--- unionfs-1.5pre-cvs200612261700/rdstate.c	2006-12-26 22:00:45.000000000 +0000
+++ unionfs-1.5pre-cvs200612261700.good/rdstate.c	2007-01-02 22:56:40.000000000 +0000
@@ -47,10 +47,8 @@
 {
 	if (!unionfs_filldir_cachep)
 		return;
-	if (kmem_cache_destroy(unionfs_filldir_cachep)) {
-		printk(KERN_ERR
-		       "unionfs_filldir_cache: not all structures were freed\n");
-	}
+	kmem_cache_destroy(unionfs_filldir_cachep);
+
 	return;
 }
 
diff -rNu unionfs-1.5pre-cvs200612261700/super.c unionfs-1.5pre-cvs200612261700.good/super.c
--- unionfs-1.5pre-cvs200612261700/super.c	2006-12-26 22:00:45.000000000 +0000
+++ unionfs-1.5pre-cvs200612261700.good/super.c	2007-01-02 22:56:16.000000000 +0000
@@ -380,9 +380,7 @@
 	print_entry_location();
 	if (!unionfs_inode_cachep)
 		goto out;
-	if (kmem_cache_destroy(unionfs_inode_cachep))
-		printk(KERN_ERR
-		       "unionfs_inode_cache: not all structures were freed\n");
+	kmem_cache_destroy(unionfs_inode_cachep);
       out:
 	print_exit_location();
 	return;
@@ -532,7 +530,7 @@
 	ino_t ino, parent_ino;
 };
 
-static int filldir(void *arg, const char *name, int len, loff_t pos, ino_t ino,
+static int filldir(void *arg, const char *name, int len, loff_t pos, u64 ino,
 		   unsigned int d_type)
 {
 	struct filldir_arg *a = arg;
